spec/models/step_spec.rb

```diff
@@ -38,4 +40,59 @@
       expect(step_item.errors).to have_key(:logic)
     end
   end
+  describe 'broken_links' do
+    it 'should return nothing if there are no link reports yet' do
+      expect(step_item.broken_links).to be_nil
+    end
+
+    it 'should return an empty array if there are link reports, but all the links work' do
+      create(:link_report, batch_id: 2, step_id: step_item.id)
+      link_checker_api_get_batch(
+        id: 2,
+        links: [
+          {
+            "uri": "https://www.gov.uk/",
+            "status": "ok",
+            "checked": "2017-04-12T18:47:16Z",
+            "errors": [],
+            "warnings": [],
+            "problem_summary": "null",
+            "suggested_fix": "null"
+          }
+        ]
+      )
+      expect(step_item.broken_links).to eql []
+    end
+
+    it 'should contain one item if there are link reports and at least one is broken' do
```

Comment:
Is this two tests hiding as one? One test to check that broken links are being returned and one test to check that the most recent link report is being returned?